### PR TITLE
Remove redundant code

### PR DESCRIPTION
**Pull request by AtariDreams**
Some of the code is either dead code, redundant, or always true. Meaning they do not do anything since the cases were already dealt with in earlier parts of the code. It would be best to remove this code.

Note that some of the code removal required a tiny bit of refactoring.

### CODE REVIEW

File: cups/auth.c [hunk 1]
```diff
@@ -330,24 +330,6 @@ _cupsSetNegotiateAuthString(
   gss_buffer_desc output_token = GSS_C_EMPTY_BUFFER;
 					/* Output token */
 
-
-  (void)method;
-  (void)resource;
-
-#  ifdef __APPLE__
```
Comment (michaelrsweet) [inline cups/auth.c:337]: This code cannot be removed as it is necessary for libcups to work on macOS when running without the Kerberos library (mainly in the installer).

File: scheduler/auth.c [hunk 2]
```diff
@@ -724,20 +724,6 @@ cupsdAuthorize(cupsd_client_t *con)	/* I - Client connection */
 					/* Output token for username */
     gss_name_t		client_name;	/* Client name */
 
-
-#  ifdef __APPLE__
-   /*
-    * If the weak-linked GSSAPI/Kerberos library is not present, don't try
-    * to use it...
-    */
-
```
Comment (michaelrsweet) [inline scheduler/auth.c:733]: Same as for cups/auth.c. Can't be removed without breaking things on macOS.

File: scheduler/colorman.c [hunk 3]
```diff
@@ -357,14 +357,6 @@ apple_register_profiles(
 			profile;	/* Current profile info dictionary */
   CFStringRef		dict_key;	/* Key in factory profile dictionary */
 
-
- /*
-  * Make sure ColorSync is available...
-  */
-
-  if (&ColorSyncRegisterDevice == NULL)
```
Comment (michaelrsweet) [inline scheduler/colorman.c:365]: Ditto.

File: scheduler/conf.c [hunk 4]
```diff
@@ -420,15 +420,6 @@ cupsdDefaultAuthType(void)
     return (default_auth_type);
 
 #ifdef HAVE_GSSAPI
-#  ifdef __APPLE__
- /*
-  * If the weak-linked GSSAPI/Kerberos library is not present, don't try
-  * to use it...
-  */
```
Comment (michaelrsweet) [inline scheduler/conf.c:427]: Ditto.

File: scheduler/ipp.c [hunk 5]
```diff
@@ -2449,18 +2449,17 @@ add_printer(cupsd_client_t  *con,	/* I - Client connection */
...
-        cupsdDeletePrinter(printer, 0);
+    send_ipp_status(con, IPP_NOT_POSSIBLE, _("Bad port-monitor \"%s\"."),
+                    attr->values[0].string.text);
+    if (!modify)
+      cupsdDeletePrinter(printer, 0);
 
-      return;
-    }
+    return;
+
```
Comment (michaelrsweet) [inline scheduler/ipp.c:2461]: Please don't introduce goto's for this. The only time we use goto is for a simple form of exception handling at the bottom of a function.

File: systemv/lpstat.c [hunk 6]
```diff
@@ -659,7 +659,7 @@ match_list(const char *list,		/* I - List of names */
 
     while (*list && !isspace(*list & 255) && *list != ',')
       list ++;
-  }
+  } while (*list);
```
Comment (michaelrsweet) [inline systemv/lpstat.c:662]: Coding style - the while should appear on a line after the closing brace...

**Review by michaelrsweet — CHANGES_REQUESTED** [review]
In general this looks good. You need to leave the NULL function checks in place since on macOS looking for the address of a function is a way to detect the presence of an optional ("weak linked") library.

**Comment by AtariDreams** [comment]
All issues addressed!

**Pull request merged**